test/common/http/http1/codec_impl_test.cc

```diff
@@ -928,6 +928,11 @@ TEST_P(Http1ServerConnectionImplTest, Http11InvalidRequest) {
 }
 
 TEST_P(Http1ServerConnectionImplTest, Http11InvalidTrailerPost) {
+  if (parser_impl_ == ParserImpl::BalsaParser) {
+    // BalsaParser only validates trailers if `enable_trailers_` is set.
+    codec_settings_.enable_trailers_ = true;
+  }
```

Comment:
This sounds reasonable to me and indeed seems to be more in the spirit of https://github.com/envoyproxy/envoy/blob/main/source/common/http/http1/codec_impl.cc#L765-L768.